credentials/generate_revocation_set.py

```diff
@@ -240,7 +240,10 @@ def generate_revocation_set_from_crl(crl_file: x509.CertificateRevocationList,
         except Exception:
             pass
 
-        serialnumber_list.append(bytes(str('{:02X}'.format(revoked_cert.serial_number)), 'utf-8').decode('utf-8'))
+        # ensuse serial number is always 2 byte aligned hex string
+        serialnumber = '{:02X}'.format(revoked_cert.serial_number)
```

Comment:
Could you explain what is going on?

I tried:

```
>>> '{:02X}'.format(1)
'01'
```

Reply:
Why not `serial_number = '%016X' % revoked_cert.serial_number` instead

Reply:
As per [rfc5280#section-4.1.2.2](https://datatracker.ietf.org/doc/html/rfc5280#section-4.1.2.2) serial number can be 20 bytes long. So what this code is trying here is just making sure the hex string is of even length and hence padding that extra zero.

Reply:
Could we add the reference to a comment?
Do we have examples with less than 19 byte long serials?

Reply:
please also explain the _why_ in the comment: why do we need this to be even? Why did we make this change (summary does not explain either ... why is this a "fix" in formatting as the old formatting looked valid as well)

Reply:
Added the comment in code and updated the description.